NorthstarDLL/core/filesystem/rpakfilesystem.cpp

```diff
@@ -255,11 +255,20 @@ void*, __fastcall, (int nPakHandle, void* pCallback))
 
 // we hook this exclusively for resolving stbsp paths, but seemingly it's also used for other stuff like vpk, rpak, mprj and starpak loads
 // tbh this actually might be for memory mapped files or something, would make sense i think
+// rtech_game.dll + 0x1e20
 // clang-format off
 HOOK(ReadFileAsyncHook, ReadFileAsync, 
 void*, __fastcall, (const char* pPath, void* pCallback))
 // clang-format on
 {
+	// NOTE [Fifty]: For some reason some users are getting pPath as null when
+	//               loading a server, ReadFileAsync uses CreateFileA and checks
+	//               its return value so this is completely safe
+	if (pPath == NULL)
+	{
+		return ReadFileAsync(pPath, pCallback);
```

Comment:
Why not just `return nullptr`? does `ReadFileAsync` do any work here that matters?

Reply:
because nullptr would be the wrong value.

Looking at a decompile, `ReadFileAsync` calls `CreateFileA` and if it returns `INVALID_HANDLE_VALUE` it returns -1 as void*
Code that relies on this would break instantly if we return anything else